### PR TITLE
Lookup instance availability for all AZs

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes https://github.com/weaveworks/eksctl/issues/5128

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by Skarlso** [comment]
Will link test run results of the two failing tests.

File: pkg/eks/api.go [hunk 1]
```diff
@@ -333,31 +335,122 @@ func ResolveAMI(ctx context.Context, provider api.ClusterProvider, version strin
...
+	instanceMap := make(map[string]map[string][]string)
+	instances := make([]string, 0)
+
+	pool := nodes.ToNodePools(spec)
+	for _, ng := range pool {
+		if _, ok := instanceMap[ng.BaseNodeGroup().Name]; !ok {
+			instanceMap[ng.BaseNodeGroup().Name] = make(map[string][]string)
+		}
+		for _, i := range ng.InstanceTypeList() {
+			instances = append(instances, i)
```
Comment (Skarlso) [inline pkg/eks/api.go:385]: This isn't the nicest, but aws deduplicates this list when it gets it.

Comment (Skarlso) [reply pkg/eks/api.go:385]: I fixed it.

**Comment by Skarlso** [comment]
```
Ran 13 of 13 Specs in 1893.771 seconds
SUCCESS! -- 13 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestDryRun (1893.77s)
PASS

```

**Comment by Skarlso** [comment]
```
Ran 2 of 2 Specs in 15.808 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestInstanceSelector (15.81s)
PASS

```

**Review by Himangini — APPROVED** [review]
👍🏻

**Review by cPu1 — APPROVED** [review]

**Comment by cPu1** [comment]
Unit tests are failing after pulling in the latest changes, we're going to have to skip this for today's release.

**Pull request merged**